components/ManageAvailabilityWizard/AvailabilityModal.tsx

```diff
@@ -0,0 +1,210 @@
+
+type Availability = {
+  [Day: string]: string[][];
+  
+}
+
+
+type AvailabilityModalProps = {
+  handleClose: () => void;
+  // handleStates: (availability: Availability, timeZone: string) => void;
+  // props for initial state values
+  initAvailability: Availability;
+  initTimeZone: string;
+  userId: string;
+  mutate: (availabiliy:any) => void;
+};
+
+const AvailabilityModal: React.FC<AvailabilityModalProps> = ({
+  handleClose,
+  // handleStates,
+  initAvailability,
+  initTimeZone,
+  userId,
+  mutate
+}) => {
+  // repeat modal states initialized by init props
+  const [availability, setAvailability] = useState(initAvailability);
+  const [timezone, setTimeZone] = useState(initTimeZone);
+  const [loading, setLoading] = useState(false)
+  const [valid, setValid] = useState(true)
+  const [errMsg, setErrMsg] = useState("")
+
+  const client = useContext(APIContext);
+
+  const sortTimes = (times: string[][]): string[][] => {
+    times.sort((t1:string[], t2:string[])=>{
+      if (t1[0] == t2[0]){
+        return 0;
+      }
+      return t1[0] < t2[0] ? -1 : 1;
+    })
+
+    return times
+  }
+  
+  const areTimesValid = (availability: Availability): [boolean, string, string] => {
+    // loops through each day
+    for(const [day, allTime] of Object.entries(availability)){
+      //makes a deepcopy so we do not change states
+      let times = JSON.parse(JSON.stringify(allTime));
+
+      // checks for overlapping times
+
+      times = sortTimes(times);
+
+
+      for(const time of times){
+        
+        let startTime = time[0];
+        let endTime = time[1];
+
+        //pad 0 in front of time if before 10am
+        if(startTime.length < 5){
+          startTime = "0" + startTime;
+        }
+        if(endTime.length < 5){
+          endTime = "0" + endTime;
+        }
+
+        // check if start time is before the end time
+        if (startTime >= endTime) {
+          return [false, day, "Start time must be before end time."]
+        }
+
+
+      }
+
+      // check if there is an overlapping time slot
+      for (let i = 1; i < times.length; i++) {
+        let prevEndTime = times[i-1][1]
+        let currStartTime = times[i][0]
+
+        //pad 0 in front of time if before 10am
+        if(currStartTime.length < 5){
+          currStartTime = "0" + currStartTime;
+        }
+        if(prevEndTime.length < 5){
+          prevEndTime = "0" + prevEndTime;
+        }
+
+        if (prevEndTime > currStartTime) {
+          console.log(prevEndTime, currStartTime)
+          return [false, day, "Times cannot overlap."]
+        }
+      }
+    }
+    return [true, "", ""]
+  }
+
+  const updateAvailabilty = (day: string, times: string[][]): void => {
+    // function to update availability state when child component changes times
+    let newAvailability: Availability = availability;
+    newAvailability[day] = times;
+
+    setAvailability({...newAvailability});
```

Comment:
when the state of a variable depends on the old state, it is bad practice to use the old availability like this. Its much better to do it like this:
`setAvailability( (oldAvail) => { // change the availibility return oldAvail;})`